core/src/main/java/org/geysermc/geyser/util/DimensionUtils.java

```diff
@@ -102,6 +102,14 @@ public static void switchDimension(GeyserSession session, String javaDimension)
         // TODO - fix this hack of a fix by sending the final dimension switching logic after sections have been sent.
         // The client wants sections sent to it before it can successfully respawn.
         ChunkUtils.sendEmptyChunks(session, player.getPosition().toInt(), 3, true);
+
+        if (BEDROCK_NETHER_ID == 2) {
+            if (bedrockDimension == BEDROCK_NETHER_ID) {
+                session.sendFog("minecraft:fog_hell");
+            } else {
+                session.removeFog("minecraft:fog_hell");
```

Comment:
This will always attempt a fog removal every dimension switch. Can we move this logic elsewhere so we don't need to do that?

Reply:
It removes it on every dimension switch so that the nether fog doesn't stay when switching to the overworld or the end. I tested on vanilla and it does infact stay unless you explicitly remove it.

Reply:
oh, i see what you mean